### DISCUSSION TITLE
FXGL WebView & IllegalStateException

### DISCUSSION THREAD

**Discussion by nicolasbulgarides** [AlmasB/FXGL#1050] [Q&A]
Hi Almas / anyone else who maybe can help...

When I try to use WebView on FXGL, I get this exception "Caused by: java.lang.IllegalStateException: Not on FX application thread; currentThread = FXGL Background Thread 4"

Now what the exceptiomn is is preetty straight forward....but is there a fix that can be done so that this does work?

**Comment by AlmasB** [comment]
Based on the exception message, I guess just run on the JavaFX Thread? `Platform.runLater()` and friends should do it.
